File: src/main/webapp/src/service/productService.js
import api from '../remote/api';

/**
 * @author Georgy Zheyts <gzheyts@gmail.com>
 */
export function queryProductPage(params) {
    return api.get('/api/products', params);
}

export function queryProductUpdate(payload) {
    return api.put('/api/products', payload);
}

export function queryProductDelete(id) {
    return api.delete(`/api/products/${id}`);
}

export function queryProductCreate(payload) {
    return api.post('/api/products', payload);
}

File: src/main/java/com/productsapp/domain/service/ProductService.java
package com.productsapp.domain.service;

import com.productsapp.domain.service.dto.PageDTO;
import com.productsapp.domain.service.dto.ProductDTO;
import com.productsapp.validation.EntityAlreadyExistsException;
import com.productsapp.validation.EntityNotFoundException;
import org.springframework.data.domain.Pageable;

/**
 * @author Georgy Zheyts <gzheyts@gmail.com>
 */
public interface ProductService {

    /**
     * Save a product.
     *
     * @param productDTO the entity to save
     * @return the persisted entity
     */
    ProductDTO save(ProductDTO productDTO) throws EntityAlreadyExistsException;

    ProductDTO update(ProductDTO productDTO) throws EntityNotFoundException;

    /**
     * Get all the products.
     *
     * @param pageable the pagination information
     * @return the list of entities
     */
    PageDTO<ProductDTO> findAll(Pageable pageable);

    /**
     * Get the "id" product.
     *
     * @param id the id of the entity
     * @return the entity
     */
    ProductDTO findById(Long id) throws EntityNotFoundException;

    /**
     * Delete the "id" product.
     *
     * @param id the id of the entity
     */
    void delete(Long id) throws EntityNotFoundException;
}
